### DISCUSSION TITLE
Resize of animated WebP breaks animation

### DISCUSSION THREAD

**Discussion by avkabanov** [ImageMagick/ImageMagick#2724] [Development]
Hi everyone, I have an animated WebP and I want to resize it, so I execute 
`$ magick convert 1.webp -resize 500x500 result.webp 
`
(initial size is 512x512)
But there is a broken animation in the resulted image. On the resulting image there is my initial image in a static mode, and another copy of my image, shifted to the right bottom corner, and it's blinking. 

Converting of gif images works fine, the problem only with WebP. How can I resize my WebP? 

```
$ magick --version
Version: ImageMagick 7.0.10-29 Q16
```
OS: Windows 10

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-layers coalesce" might help.

Please show a sample input.

**Reply by avkabanov** [reply]
That helped! 

But the size of the image increased, considering I've decreased the resolution! I've tried to use -deconstruct or -layers CompareAny, but it gave no result 
`$ magick 1.webp -layers coalesce -resize 500x500 -layers CompareAny result.webp`

I've updated the information about IM and OS. Please let me know if you still need the sample input

**Comment by snibgo** [comment]
Resizing almost certainly adds new colours, which probably reduces the effectiveness of compression.

If you don't post the input, I can't experiment to try to solve your problem.

**Reply by avkabanov** [reply]
WebP is not supported as attached file format, so I've zipped it. There are two files: input and output. 
initial resolution: 220x220, I've resized it to 200x200, 
but the size increased from 12.6KB till 15.9KB 
`magick input.webp -layers coalesce -resize 200x200 output.webp
`
[input.zip](https://github.com/ImageMagick/ImageMagick/files/5386249/input.zip)

Thanks in advance!

**Comment by avkabanov** [comment]
I found a similar topic for gif, but with IM6. They proposed to use -deconstruct after resizing. In the context of gif - deconstruct really reduced the image size. But not for WebP

https://askubuntu.com/questions/257831/how-can-i-resize-an-animated-gif-file-using-imagemagick

**Comment by snibgo** [comment]
There are many options for webp. See https://imagemagick.org/script/webp.php

For example, using your input.webp:
```
magick input.webp -layers coalesce -resize 200x200 -quality 40 -define webp:method=6 output2.webp
```
The output is smaller than the input.

(Tested with IM v7.0.10-27 on Windows 8.1.)

**Reply by avkabanov** [reply]
That's not exactly the same, because you additionally reduce the quality of the picture
